### ISSUE TITLE
Doc error at http://docs.scipy.org/doc/numpy-1.10.1/reference/generated/numpy.histogram2d.html

### ISSUE THREAD

**Issue by spicybaguette** [numpy/numpy#7538]
![figure_1](https://cloud.githubusercontent.com/assets/18401228/14433291/0b7c14c8-000e-11e6-8861-c723f5583e70.png)

In the demonstration of numpy.histogram2d, H is calculated using xedges as the vertical bins (index 0) and y as the vertical variable :  
`H, xedges, yedges = np.histogram2d(y, x, bins=(xedges, yedges))`

However imshow is called using xedges as the horizontal limits : 
`im = plt.imshow(H, interpolation='nearest', origin='low', extent=[xedges[0], xedges[-1], yedges[0], yedges[-1]])`

when it should be 
`im = plt.imshow(H, interpolation='nearest', origin='low',  extent=[yedges[0], yedges[-1], xedges[0], xedges[-1]])`

This will change the axes with respect to the data provided. Since imshow simply places the numerical limits set by the user next to the image data, this error is rather difficult to demonstrate, especially when symmetric distributions are used on both the x and y axes. Using a skewed distribution and comparing to hexbin allowed me to double check the data shown by histogram2d.

It might be a good idea to take one of the following actions : 

Change the definition of numpy.histogram2 at the top of the page from
`numpy.histogram2d(x, y, bins=10, range=None, normed=False, weights=None)`
to 
`numpy.histogram2d(y,x, bins=10, range=None, normed=False, weights=None)`
or use 
`H, xedges, yedges = np.histogram2d(x, y, bins=(xedges, yedges))`
or 
`H, yedges, xedges = np.histogram2d(y, x, bins=(yedges, xedges))`
in order to keep y as the vertical coordinate. 

code used to generate example (run with python 2.7.9) : 

```
import numpy as np 
import matplotlib as mpl
import matplotlib.pyplot as plt

x = np.random.gamma(3.0,scale=3.0, size=10000) 
y = np.random.normal(1, 1, 10000)

xedges = np.linspace(-1,1,num=10)
yedges = np.linspace(1,9,num=10)

print(xedges)
print(yedges)

H, xedges, yedges = np.histogram2d(y, x, bins=(xedges, yedges))

fig = plt.figure(figsize=(7, 3))
ax = fig.add_subplot(131)
ax.set_title('imshow: equidistant')

cx1 = mpl.cm.get_cmap('rainbow')
# Plot H as shown in example 
im = plt.imshow(H, interpolation='nearest',origin='low',extent=[xedges[0],xedges[-1],yedges[0],yedges[-1]],cmap=cx1,aspect="auto")

# Plot hexbin using same parameters
ax = fig.add_subplot(132)
ax.set_title('hexbin')
ax.hexbin(y,x,gridsize=50, cmap=cx1)
ax.set_aspect('auto')

ax.set_xlim([xedges[0],xedges[-1]])
ax.set_ylim([yedges[0],yedges[-1]])

plt.show()
```

**Comment by wrwrwr** [comment]
@spicybaguette I believe this has been fixed in 2a86f356db2cb061b08be03813b77685dd149e2a. Please take a look at the [current version](https://github.com/numpy/numpy/files/557243/numpy-ref-1141.pdf) and see if anything more should be changed.

**Comment by mattip** [comment]
Closing The reporter did not respond to verify the fix. If there are still inconsistencies, please open a new issue.
